lib/src/ansi_code.dart

```diff
@@ -0,0 +1,278 @@
+  static const backgroundRed =
+      const AnsiCode._('red', AnsiCodeType.background, 41);
+  static const backgroundGreen =
+      const AnsiCode._('green', AnsiCodeType.background, 42);
+  static const backgroundYellow =
+      const AnsiCode._('yellow', AnsiCodeType.background, 43);
+  static const backgroundBlue =
+      const AnsiCode._('blue', AnsiCodeType.background, 44);
+  static const backgroundMagenta =
+      const AnsiCode._('magenta', AnsiCodeType.background, 45);
+  static const backgroundCyan =
+      const AnsiCode._('cyan', AnsiCodeType.background, 46);
+  static const backgroundLightGray =
+      const AnsiCode._('light gray', AnsiCodeType.background, 47);
+  static const backgroundDefault =
+      const AnsiCode._('default', AnsiCodeType.background, 49);
+  static const backgroundDarkGray =
+      const AnsiCode._('dark gray', AnsiCodeType.background, 100);
+  static const backgroundLightRed =
+      const AnsiCode._('light red', AnsiCodeType.background, 101);
+  static const backgroundLightGreen =
+      const AnsiCode._('light green', AnsiCodeType.background, 102);
+  static const backgroundLightYellow =
+      const AnsiCode._('light yellow', AnsiCodeType.background, 103);
+  static const backgroundLightBlue =
+      const AnsiCode._('light blue', AnsiCodeType.background, 104);
+  static const backgroundLightMagenta =
+      const AnsiCode._('light magenta', AnsiCodeType.background, 105);
+  static const backgroundLightCyan =
+      const AnsiCode._('light cyan', AnsiCodeType.background, 106);
+  static const backgroundWhite =
+      const AnsiCode._('white', AnsiCodeType.background, 107);
+
+  /// The numeric value associated with this code.
+  final int code;
+
+  /// A description of this code.
+  final String name;
+
+  /// The type of code that is represented.
+  final AnsiCodeType type;
+
+  const AnsiCode._(this.name, this.type, this.code);
+
+  /// Represents the value escaped for use in terminal output.
+  String get escape => "\x1B[${code}m";
+
+  /// Wraps [value] with the [escape] value for this code, followed by
+  /// [resetAll].
+  ///
+  /// Returns `value` unchanged if
+  ///   * [value] is `null` or empty
+  ///   * `dart:io` `stdout.supportsAnsiEscapes` is `false`
+  ///   * [type] is [AnsiCodeType.reset]
+  String wrap(String value) => (io.stdout.supportsAnsiEscapes &&
+          type != AnsiCodeType.reset &&
+          value != null &&
+          value.isNotEmpty)
+      ? "$escape$value${resetAll.escape}"
+      : value;
+
+  /// Returns a [String] formatted with the codes provided in [foreground],
+  /// [background], and [style] or [styles].
+  ///
+  /// Returns `value` unchanged if
+  ///   * [value] is `null` or empty
+  ///   * `dart:io` `stdout.supportsAnsiEscapes` is `false`
+  ///   * All of [foreground], [background], [style], or [styles] are `null` or
+  ///     empty.
+  ///
+  /// Throws an [ArgumentError] if by [style] and [styles] are provided.
+  static String wrapWith(String value,
```

Comment:
I guess this could just take an `Iterable<AnsiCode>` and call it a day.

Thoughts, @matanlurey ?